src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      currentState = {};
+    } else if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      const newState = {};
+
+      for (const key in currentState) {
+        if (!action.keysToRemove.includes(key)) {
+          newState[key] = currentState[key];
+        }
+      }
+      currentState = newState;
+    }
```

Comment:
Include a default case in your switch statement to handle any unexpected action types. This is important for error handling and to ensure that the function behaves predictably even with incorrect input.